.travis.yml

```diff
@@ -30,69 +49,57 @@ matrix:
       env: TF_PACKAGE=tensorflow==1.1.0 KERAS_PACKAGE=keras==2.0.0 MPI=MPICH
     - python: "3.6"
       env: TF_PACKAGE=tensorflow==1.1.0 KERAS_PACKAGE=keras==2.0.0 MPI=MPICH
-    - python: "3.6"
+    - python: "3.4"
       env: TF_PACKAGE=tf-nightly KERAS_PACKAGE=git+https://github.com/fchollet/keras.git MPI=OpenMPI
 
 install:
   - |
-    if [ $MPI == "OpenMPI" ]; then
-      wget -O /tmp/openmpi-3.0.0-bin.tar.gz https://github.com/uber/horovod/files/1439728/openmpi-3.0.0-bin.tar.gz
-      pushd /usr/local && sudo tar -zxf /tmp/openmpi-3.0.0-bin.tar.gz && popd
-      export PATH=/usr/local/openmpi/bin:$PATH
+    if [[ $MPI == "OpenMPI" ]]; then
+      docker exec $CONTAINER /bin/sh -c "wget -O /tmp/openmpi-3.0.0-bin.tar.gz https://github.com/uber/horovod/files/1596799/openmpi-3.0.0-bin.tar.gz"
+      docker exec $CONTAINER /bin/sh -c "cd /usr/local && tar -zxf /tmp/openmpi-3.0.0-bin.tar.gz && ldconfig"
     else
       # installs mpich version 3.0.4
-      sudo apt-get install mpich
+      docker exec $CONTAINER /bin/sh -c "apt-get install -y mpich"
     fi
 
   # TensorFlow
-  - if [[ $TRAVIS_PYTHON_VERSION == 2* ]]; then pip install "$TF_PACKAGE"; fi
-  - if [[ $TRAVIS_PYTHON_VERSION == 3* ]]; then pip3 install "$TF_PACKAGE"; fi
+  - docker exec $CONTAINER /bin/sh -c "pip install $TF_PACKAGE"
```

Comment:
so the force upgrade, upgrades pip to pip3 incase of python3? Hence we don't have to do explicit pip3 installs for python3.x?

Reply:
Purpose of force upgrade is to create `pip` binary even for Python3, which `apt install python3-pip` fails to do.